### ISSUE TITLE
GGPO_ERRORCODE_PREDICTION_THRESHOLD after every ggpo_add_local_input 

### ISSUE THREAD

**Issue by AndreaJens** [pond3r/ggpo#60]
I apologize if this might be a stupid question, but I am having troubles in implementing this library in a sample project of mine. 
While running the vectorwar demo, everything seems to work fine, without issues. I can compile and execute the game without troubles and the synchronization via GGPO works flawlessly.
However, when I try to implement GGPO in my own project, after the connection between the players has been established (**GGPO_EVENTCODE_RUNNING** has been issued), whenever I call a _ggpo_add_local_input_ command, it always fails with the above return value (**GGPO_ERRORCODE_PREDICTION_THRESHOLD**).
Is there any step I have to perform between the first time I receive a GGPO_EVENTCODE_RUNNING and the first _ggpo_add_local_input_ / _ggpo_synchronize_input_ that I am missing?
If not, what could be the cause of this error?

To summarize:
- I start the session with _ggpo_start_session_ (result = GGPO_ERRORCODE_SUCCESS)
- add the two players with _ggpo_add_player_ (result = GGPO_ERRORCODE_SUCCESS)
- call for _ggpo_advance_frame_ (result = GGPO_ERRORCODE_SUCCESS) at 60fps. Otherwise, call _ggpo_idle_
- the two players connect and I get the events GGPO_EVENTCODE_SYNCHRONIZED_WITH_PEER, then GGPO_EVENTCODE_RUNNING
- after this, whenever I call _ggpo_add_local_input_, it fails with result GGPO_ERRORCODE_PREDICTION_THRESHOLD, thus preventing synchronization between the clients.

The above issue happens even when running two instances of the game on the same machine, on different ports.

**Comment by AndreaJens** [comment]
Found the issue, it was due to me calling _ggpo_advance_frame_ right after the input update even if the _ggpo_add_local_input_ was unsuccessful.
